contents/javascript/Javascript-promises.md

```diff
@@ -89,58 +87,117 @@ These situations correspond to the 3 states of Promises in Javascript.
 
 Promises provide the ability to specify how the execution of some part of your code would depend on the status of an asynchronous operation. It can now wait for the asynchronous operation to resolve first before doing any work on its result.
 
-## Quickstart
+## Getting Started 
+
+Here, we will be showing you some sample code to get started with promises. Let's learn how to go from callbacks to promises using a few examples.
+
+### From callbacks to promises
+Say we have two <tooltip content="requires a long time to complete">time intensive</tooltip> functions, `getData()` and `filterData()` which require some time to complete. You will have to get the data from some server using `getData()`, then process it using `filterData()`, before you can start displaying the results.
```

Comment:
Do you even need this term _time intensive_? The sentence works even without it.
>Say we have two functions, `getData()` and `filterData()` which require some time to complete.